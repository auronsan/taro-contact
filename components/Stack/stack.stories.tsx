import React from 'react';
import { Stack } from './index';

export default {
  title: 'Components/Stack',
  component: Stack,
};

export const Default = () => (
  <Stack>
    <div>Item 1</div>
    <div>Item 2</div>
    <div>Item 3</div>
  </Stack>
);

export const JustifyCenter = () => (
  <Stack justify="center">
    <div>Item 1</div>
    <div>Item 2</div>
    <div>Item 3</div>
  </Stack>
);

export const JustifySpaceBetween = () => (
  <Stack justify="space-between">
    <div>Item 1</div>
    <div>Item 2</div>
    <div>Item 3</div>
  </Stack>
);

export const JustifyEnd = () => (
  <Stack justify="end">
    <div>Item 1</div>
    <div>Item 2</div>
    <div>Item 3</div>
  </Stack>
);

export const Wrap = () => (
  <Stack wrap="wrap">
    <div>Item 1</div>
    <div>Item 2</div>
    <div>Item 3</div>
  </Stack>
);

export const NoWrap = () => (
  <Stack wrap="nowrap">
    <div>Item 1</div>
    <div>Item 2</div>
    <div>Item 3</div>
  </Stack>
);

export const GapXs = () => (
  <Stack gap="xs">
    <div>Item 1</div>
    <div>Item 2</div>
    <div>Item 3</div>
  </Stack>
);

export const GapSm = () => (
  <Stack gap="sm">
    <div>Item 1</div>
    <div>Item 2</div>
    <div>Item 3</div>
  </Stack>
);

export const GapMd = () => (
  <Stack gap="md">
    <div>Item 1</div>
    <div>Item 2</div>
    <div>Item 3</div>
  </Stack>
);

export const GapLg = () => (
  <Stack gap="lg">
    <div>Item 1</div>
    <div>Item 2</div>
    <div>Item 3</div>
  </Stack>
);

export const GapXl = () => (
  <Stack gap="xl">
    <div>Item 1</div>
    <div>Item 2</div>
    <div>Item 3</div>
  </Stack>
);

export const Padding = () => (
  <Stack p={16}>
    <div>Item 1</div>
    <div>Item 2</div>
    <div>Item 3</div>
  </Stack>
);
