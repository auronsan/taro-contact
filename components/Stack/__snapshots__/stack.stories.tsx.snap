// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Stack Default smoke-test 1`] = `
<main class="roboto-normal">
  <div class="stack_root__KVCbK stack_gap-sm__rrtkk"
       style="padding: 0px;"
  >
    <div>
      Item 1
    </div>
    <div>
      Item 2
    </div>
    <div>
      Item 3
    </div>
  </div>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;

exports[`Components/Stack GapLg smoke-test 1`] = `
<main class="roboto-normal">
  <div class="stack_root__KVCbK stack_gap-lg__eTmHP"
       style="padding: 0px;"
  >
    <div>
      Item 1
    </div>
    <div>
      Item 2
    </div>
    <div>
      Item 3
    </div>
  </div>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;

exports[`Components/Stack GapMd smoke-test 1`] = `
<main class="roboto-normal">
  <div class="stack_root__KVCbK stack_gap-md__l1Cy_"
       style="padding: 0px;"
  >
    <div>
      Item 1
    </div>
    <div>
      Item 2
    </div>
    <div>
      Item 3
    </div>
  </div>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;

exports[`Components/Stack GapSm smoke-test 1`] = `
<main class="roboto-normal">
  <div class="stack_root__KVCbK stack_gap-sm__rrtkk"
       style="padding: 0px;"
  >
    <div>
      Item 1
    </div>
    <div>
      Item 2
    </div>
    <div>
      Item 3
    </div>
  </div>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;

exports[`Components/Stack GapXl smoke-test 1`] = `
<main class="roboto-normal">
  <div class="stack_root__KVCbK stack_gap-xl__Nmb1t"
       style="padding: 0px;"
  >
    <div>
      Item 1
    </div>
    <div>
      Item 2
    </div>
    <div>
      Item 3
    </div>
  </div>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;

exports[`Components/Stack GapXs smoke-test 1`] = `
<main class="roboto-normal">
  <div class="stack_root__KVCbK stack_gap-xs__xTnny"
       style="padding: 0px;"
  >
    <div>
      Item 1
    </div>
    <div>
      Item 2
    </div>
    <div>
      Item 3
    </div>
  </div>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;

exports[`Components/Stack JustifyCenter smoke-test 1`] = `
<main class="roboto-normal">
  <div class="stack_root__KVCbK stack_justify-center__CmTLW stack_gap-sm__rrtkk"
       style="padding: 0px;"
  >
    <div>
      Item 1
    </div>
    <div>
      Item 2
    </div>
    <div>
      Item 3
    </div>
  </div>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;

exports[`Components/Stack JustifyEnd smoke-test 1`] = `
<main class="roboto-normal">
  <div class="stack_root__KVCbK stack_justify-end___HBdD stack_gap-sm__rrtkk"
       style="padding: 0px;"
  >
    <div>
      Item 1
    </div>
    <div>
      Item 2
    </div>
    <div>
      Item 3
    </div>
  </div>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;

exports[`Components/Stack JustifySpaceBetween smoke-test 1`] = `
<main class="roboto-normal">
  <div class="stack_root__KVCbK stack_justify-space-between__v6A4W stack_gap-sm__rrtkk"
       style="padding: 0px;"
  >
    <div>
      Item 1
    </div>
    <div>
      Item 2
    </div>
    <div>
      Item 3
    </div>
  </div>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;

exports[`Components/Stack NoWrap smoke-test 1`] = `
<main class="roboto-normal">
  <div class="stack_root__KVCbK stack_stack-nowrap__DhAIU stack_gap-sm__rrtkk"
       style="padding: 0px;"
  >
    <div>
      Item 1
    </div>
    <div>
      Item 2
    </div>
    <div>
      Item 3
    </div>
  </div>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;

exports[`Components/Stack Padding smoke-test 1`] = `
<main class="roboto-normal">
  <div class="stack_root__KVCbK stack_gap-sm__rrtkk"
       style="padding: 16px;"
  >
    <div>
      Item 1
    </div>
    <div>
      Item 2
    </div>
    <div>
      Item 3
    </div>
  </div>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;

exports[`Components/Stack Wrap smoke-test 1`] = `
<main class="roboto-normal">
  <div class="stack_root__KVCbK stack_stack-wrap__OMIUk stack_gap-sm__rrtkk"
       style="padding: 0px;"
  >
    <div>
      Item 1
    </div>
    <div>
      Item 2
    </div>
    <div>
      Item 3
    </div>
  </div>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;
